### PR TITLE
[master] RI-90 Create ironic config script for gating

### PR DESCRIPTION
**Pull request by cjloader**
This reverts the previous patch into master.

This commit provides the steps required for gating to be able to
run tests against a fully configured ironic deployment.

(cherry picked from commit ec49562)

Issue: [RI-90](https://rpc-openstack.atlassian.net/browse/RI-90)

### CODE REVIEW

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_with_swift

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_swift

**Comment by odyssey4me** [comment]
recheck_cit_xenial_deploy_swift

**Comment by cjloader** [comment]
tempest image failure, looking into it

File: scripts/bootstrap-aio.yml [hunk 1]
```diff
@@ -158,8 +158,8 @@
   include: "{{ lookup('env', 'OA_DIR') }}/tests/bootstrap-aio.yml"
   vars:
     bootstrap_host_apt_distribution_suffix_list: "{{ (lookup('env', 'RPCO_APT_ARTIFACTS_AVAILABLE') | bool) | ternary([], ['updates', 'backports']) }}"
-    bootstrap_host_scenario: "{% if lookup('env', 'DEPLOY_MAGNUM') == 'yes' %}magnum{% elif lookup('env', 'DEPLOY_IRONIC') == 'yes' %}ironic{% else %}{{ lookup('env', 'DEPLOY_CEPH') |bool |ternary('ceph','swift') }}{%endif%}"
```
Comment (major) [inline scripts/bootstrap-aio.yml]: Why are we removing `DEPLOY_MAGNUM` in this commit?

Comment (cjloader) [reply scripts/bootstrap-aio.yml]: I thought we REMOVED magnum.

Comment (cjloader) [reply scripts/bootstrap-aio.yml]: Newton did

Comment (mendeni) [reply scripts/bootstrap-aio.yml]: Magnum is deprecated, but doesn't need to be removed in this commit.
Let's leave `DEPLOY_MAGNUM` alone and address removal in a separate PR.

Comment (prometheanfire) [reply scripts/bootstrap-aio.yml]: True, though if we don't do it here we should document it for removal later.

Comment (cjloader) [reply scripts/bootstrap-aio.yml]: addressed.

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_swift

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_swift

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_swift

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_swift

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_swift

File: scripts/bootstrap-aio.yml (hunk 1 above)
Comment (odyssey4me) [reply scripts/bootstrap-aio.yml]: Actually, ``DEPLOY_MAGNUM`` was added by https://github.com/rcbops/rpc-openstack/commit/e2fdc5989f236da5d3e1135b83bf0c4b7dc8c6b2 mistakenly, so it should be removed in this PR.

**Comment by cjloader** [comment]
recheck_cit_xenial_deploy_swift

**Comment by odyssey4me** [comment]
It looks like this is failing on this known issue: https://rpc-openstack.atlassian.net/browse/RO-2781

File: scripts/bootstrap-aio.yml [hunk 2]
```diff
@@ -158,8 +158,8 @@
...
     bootstrap_user_variables_template: "user_variables.aio.yml.j2"
+    bootstrap_host_scenario: "{% if lookup('env', 'DEPLOY_MAGNUM') == 'yes' %}magnum{% elif lookup('env', 'DEPLOY_IRONIC') == 'yes' %}ironic{% else %}{{ lookup('env', 'DEPLOY_CEPH') |bool |ternary('ceph','swift') }}{%endif%}"
```
Comment (odyssey4me) [inline scripts/bootstrap-aio.yml]: ``DEPLOY_MAGNUM`` is still here, and should not be.

File: scripts/bootstrap-aio.yml (hunk 1 above)
Comment (cjloader) [reply scripts/bootstrap-aio.yml]: https://github.com/rcbops/rpc-openstack/blob/newton/scripts/bootstrap-aio.yml

it is not in newton

**Comment by cjloader** [comment]
@odyssey4me addressed

**Pull request merged**